boost-cmake/CMakeLists.txt

```diff
@@ -0,0 +1,104 @@
+cmake_minimum_required(VERSION 3.12)
+project(Boost-CMake)
+
+option(BOOST_DISABLE_TESTS "Do not build test targets, even if building standalone" OFF)
+
+set(BOOST_URL "https://boostorg.jfrog.io/artifactory/main/release/1.71.0/source/boost_1_71_0.tar.bz2" CACHE STRING "Boost download URL")
+set(BOOST_URL_SHA256 "d73a8da01e8bf8c7eda40b4c84915071a8c8a0df4a6734537ddde4a8580524ee" CACHE STRING "Boost download URL SHA256 checksum")
+
+include(FetchContent)
+FetchContent_Declare(
+  Boost
+  URL ${BOOST_URL}
+  URL_HASH SHA256=${BOOST_URL_SHA256}
+)
+FetchContent_GetProperties(Boost)
+
+if(NOT Boost_POPULATED)
+  message(STATUS "Fetching Boost")
+  FetchContent_Populate(Boost)
+  message(STATUS "Fetching Boost - done")
+  set(BOOST_SOURCE ${boost_SOURCE_DIR})
+endif()
+
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules)
+include(CheckBoostVersion)
+
+message(STATUS "Boost found: ${BOOST_VERSION} ${BOOST_SOURCE}")
+
+include(StandaloneBuild)
+include(PlatformDetect)
+include(AddBoostLib)
+include(AddBoostTest)
+
+set_property(GLOBAL PROPERTY USE_FOLDERS TRUE)
+
+if(USE_ANDROID)
```

Comment:
Do we need this? Is someone ever going to run this on android?

Reply:
everything in `boost-cmake` is subtreed in from https://github.com/Orphis/boost-cmake and we shouldn't change it locally